core/chains/evm/txmgr/txmgr.go

```diff
@@ -242,36 +286,107 @@ func (b *Txm) runLoop(eb *EthBroadcaster, ec *EthConfirmer) {
 	ctx, cancel := utils.ContextFromChan(b.chStop)
 	defer cancel()
 
+	var stopped bool
+
+	execReset := func(r *reset) {
```

Comment:
Some tricky concurrency in here, lots of moving parts. Pls review carefully to check for potential mistakes.

Reply:
Could this be a helper func/method rather than a closure? Might be easier to read.

Reply:
Would be messy because it closes over `eb`, `ec` and `stopped`